### PR TITLE
Increase unit test code coverage and fix sonar code smells

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#16661](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/16661)

## Description

- Increase unit test code coverage
- Fix sonar cloud code smells
- Removed commented out code for immunization service tests as code coverage is  100%.

https://sonarcloud.io/code?id=bcgov_healthgateway&selected=bcgov_healthgateway%3AApps%2FImmunization

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## UI Changes

None

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Immunization/test/unit/Controllers.Test/ImmunizationControllerTests.cs [hunk 1]
```diff
@@ -43,47 +42,47 @@ public class ImmunizationControllerTests
...
-                            Immunization = new ImmunizationDefinition
-                            {
-                                Name = "Mocked Name",
-                                ImmunizationAgents = new List<ImmunizationAgent>(),
-                            },
-                        },
-                    },
-                    new List<ImmunizationRecommendation>()),
+                    [event1, event2],
+                    []),
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Controllers.Test/ImmunizationControllerTests.cs:85]: Nit: now that these lines are much shorter, they can probably all fit on a single line.

File: Apps/Immunization/test/unit/Delegates.Test/ImmunizationDelegateTests.cs [hunk 2]
```diff
@@ -89,9 +88,8 @@ public async Task GetImmunizations()
             PhsaResult<ImmunizationResponse> phsaResponse = new()
             {
                 Result = new ImmunizationResponse(
-                    new List<ImmunizationViewResponse>
-                        { expectedViewResponse },
-                    new List<ImmunizationRecommendationResponse>()),
+                    [expectedViewResponse],
+                    []),
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Delegates.Test/ImmunizationDelegateTests.cs:92]: Nit: now that these lines are much shorter, they can probably all fit on a single line.

File: Apps/Immunization/test/unit/Delegates.Test/ImmunizationDelegateTests.cs [hunk 3]
```diff
@@ -145,9 +143,8 @@ public async Task GetImmunizationsThrowsException()
...
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Delegates.Test/ImmunizationDelegateTests.cs:147]: Nit: now that these lines are much shorter, they can probably all fit on a single line.

File: Apps/Immunization/test/unit/Services.Test/ImmunizationServiceTests.cs [hunk 4]
```diff
@@ -73,17 +73,16 @@ public async Task ShouldGetImmunizations(bool canAccessDataSource)
...
                             {
                                 Id = Guid.NewGuid(),
                                 Name = "MockImmunization",
                                 OccurrenceDateTime = DateTime.SpecifyKind(DateTime.Now, DateTimeKind.Unspecified),
                                 SourceSystemId = "MockSourceID",
                             },
-                        },
-                        new List<ImmunizationRecommendationResponse>()),
+                        ],
+                        []),
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Services.Test/ImmunizationServiceTests.cs:85]: Nit: now that these lines are much shorter, they can probably all fit on a single line.

File: Apps/Immunization/test/unit/Services.Test/ImmunizationServiceTests.cs [hunk 5]
```diff
@@ -627,69 +284,15 @@ private static RequestResult<PhsaResult<ImmunizationResponse>> GetPhsaResult(Imm
                 {
                     LoadState = new() { RefreshInProgress = false },
                     Result = new(
-                        new List<ImmunizationViewResponse>(),
-                        new List<ImmunizationRecommendationResponse>
-                        {
-                            immzRecommendationResponse,
-                        }),
+                        [],
+                        [immzRecommendationResponse]),
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Services.Test/ImmunizationServiceTests.cs:288]: Nit: now that these lines are much shorter, they can probably all fit on a single line.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/Immunization/test/unit/Controllers.Test/ImmunizationControllerTests.cs (hunk 1 above)
Comment (BrianMaki) [reply Apps/Immunization/test/unit/Controllers.Test/ImmunizationControllerTests.cs:85]: Need to configure Rider.  Change next time.

File: Apps/Immunization/test/unit/Delegates.Test/ImmunizationDelegateTests.cs (hunk 2 above)
Comment (BrianMaki) [reply Apps/Immunization/test/unit/Delegates.Test/ImmunizationDelegateTests.cs:92]: Need to configure Rider.  Change next time.

File: Apps/Immunization/test/unit/Services.Test/ImmunizationServiceTests.cs (hunk 4 above)
Comment (BrianMaki) [reply Apps/Immunization/test/unit/Services.Test/ImmunizationServiceTests.cs:85]: Need to configure Rider.  Change next time.

File: Apps/Immunization/test/unit/Services.Test/ImmunizationServiceTests.cs (hunk 5 above)
Comment (BrianMaki) [reply Apps/Immunization/test/unit/Services.Test/ImmunizationServiceTests.cs:288]: Need to configure Rider.  Change next time.

File: Apps/Immunization/test/unit/Delegates.Test/ImmunizationDelegateTests.cs (hunk 3 above)
Comment (BrianMaki) [reply Apps/Immunization/test/unit/Delegates.Test/ImmunizationDelegateTests.cs:147]: Need to configure Rider.  Change next time.

**Pull request merged**